### ISSUE TITLE
[BUG][Harbor] Failed to create harbor project & warning message when copy image

### ISSUE THREAD

**Issue by STARRY-S** [cnrancher/hangar#44]
## Description
- Hangar version: `v1.8.0`, `v1.7.3`

Failed to create harbor project automatically when loading images, and warning message appears when loading image to harbor registry server:

```console
$ hangar load -s ./saved-images.zip -d h2.example.com:30080 --tls-verify=false
[10:18:26] [INFO] Arch List: [amd64,arm64]
[10:18:26] [INFO] OS List: [linux]
[10:18:27] [WARN] Failed, retrying in 100ms ... (1/3). Error: reading manifest 3 in h2.example.com:30080/library/alpine: unknown: artifact library/alpine:3 not found
[10:18:28] [WARN] Failed, retrying in 100ms ... (2/3). Error: reading manifest 3 in h2.example.com:30080/library/alpine: unknown: artifact library/alpine:3 not found
[10:18:29] [WARN] Failed, retrying in 100ms ... (3/3). Error: reading manifest 3 in h2.example.com:30080/library/alpine: unknown: artifact library/alpine:3 not found
[10:18:30] [INFO] [IMG:1] Loading [docker.io/library/alpine:3]...
```

**Comment by STARRY-S** [comment]
## Reason

1. Bug at [here](https://github.com/cnrancher/hangar/blob/v1.8.0/pkg/harbor/harbor.go#L50C3-L59C4) caused failed to create harbor project automatically, affected version `v1.7.2 - v1.8.0`.
1. Harbor responses non-standard not found error code and c/image library cannot detect it.
    Will add workaround to detect error string content in the [retry function](https://github.com/cnrancher/hangar/blob/v1.8.1-rc1/pkg/image/internal/private/private.go#L19C3-L32C15).

----

Will fix in both `v1.8.1` and `v1.7.4`.

**Comment by STARRY-S** [comment]
Hangar validation test for distribution registry & Harbor has intergrated into CI Pipeline:
https://drone-pandaria.cnrancher.com/cnrancher/hangar/454/1/3
